### PR TITLE
[MTG-758] [MTG-802] Rectify close_voter ix

### PR DESCRIPTION
**Pull request by kstepanovdev**
To close `voter` and its `ATA`  an end-user need to pass in all `ATA` gathered from mints of a `Registrar`; even if those `ATA` don't exist, their addresses may still be calculated.

### CODE REVIEW

**Review by StanChe — COMMENTED** [review]
With this change we definitely need to document, what should be sent. It's quite counterintuitive to pass in all the possible associated token accounts. Also extending the number of voting mints might make closing the registrar nearly impossible, but that's smth we should just write in comments. Also I'd say we should restrict the positions of the accounts being closed. So we have several voting mints that are not zeroed. And the first remaining account should be for the first non-zeroed voting mint. The second - for the second and so on. This way may have a linear complexity instead of quadratic.

File: programs/voter-stake-registry/src/instructions/close_voter.rs [hunk 1]
```diff
@@ -97,31 +111,42 @@ pub fn close_voter<'info>(ctx: Context<'_, '_, '_, 'info, CloseVoter<'info>>) ->
...
-        for deposit_vault_info in ctx.remaining_accounts {
-            let deposit_vault_ta = Account::<TokenAccount>::try_from(deposit_vault_info)
-                .map_err(|_| MplStakingError::DeserializationError)?;
-            registrar.voting_mint_config_index(deposit_vault_ta.mint)?;
+        let calculated_atas_to_close = filtered_mints.into_iter().map(|voting_mint_config| {
+            get_associated_token_address(&ctx.accounts.voter.key(), &voting_mint_config.mint)
+        });
 
+        for (index, calculated_ata_to_close) in calculated_atas_to_close.enumerate() {
+            require!(
```
Comment (StanChe) [inline programs/voter-stake-registry/src/instructions/close_voter.rs]: We've already checked this before. What might have changed?

**Review by StanChe — APPROVED** [review]
Great job! Looks like some redundant checks are there.

File: programs/voter-stake-registry/src/instructions/close_voter.rs (hunk 1 above)
Comment (kstepanovdev) [reply programs/voter-stake-registry/src/instructions/close_voter.rs]: Tunnel vision. Thanks!

**Pull request merged**